### DISCUSSION TITLE
Error 4: Undefined variable liquidsoap

### DISCUSSION THREAD

**Discussion by mykauf** [savonet/liquidsoap#2117] [General]
What can I do? I installed (Ubuntu Server 20.04) liquidsoap and it get this error. The code looks like this.
`#!/usr/bin/liquidsoap

liquidsoap 'output(playlist("/home/john/stream/nt-playlist.txt"))'`

Thank you for your help.

**Comment by toots** [comment]
Hi,

You have several ways of running a script:

* You can call the binary directly from the shell:
```
% liquidsoap 'output(playlist("/home/john/stream/nt-playlist.txt"))'
```

* You can put the code into a script file, say `/path/to/script.liq` that looks like that:
```ruby
output(playlist("/home/john/stream/nt-playlist.txt"))
```
And then execute it with the binary:
```
% liquidsoap /path/to/script.liq
```

* Or you can write the same file `/path/to/script.liq` but with the script shebang:
```
#!/usr/bin/env liquidsoap

output(playlist("/home/john/stream/nt-playlist.txt"))
```
Make it executable:
```
% chmod +x /path/to/script.liq
```
And execute it directly:
```
% /path/to/script.liq
```
For for detailed instructions, I highly recommend checking out the liquidsoap book: https://www.liquidsoap.info/doc-dev/book.html

**Comment by mykauf** [comment]
Thank you very much! It works!
